### PR TITLE
Edited readme file with common architecture examples

### PR DESCRIPTION
**Pull request by jimni1222**
## Proposed changes

This PR introduces modification of README.md file with common archituecture functions.

For link, Klaytn Docs PRs are not applied yet, so i will merge this PR after Klaytn Docs PRs are merged (to check link works or not).

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-js/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-js)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

refer #249 

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,112 +67,79 @@ $ node
...
 ```
 
-## Using caver-js account/wallet
-You can easily manage your account by using the account / wallet packages provided by caver-js.
-[caver.klay.accounts](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.klay.accounts) package provides functions related to accounts, such as create, signTransaction, and privateKeyToAccount.
-[caver.klay.accounts.wallet](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.klay.accounts#wallet) provides the **in-memory wallet** for easy account management in caver-js.
+## Using caver-js keyring/wallet
 
-**Note** Functions associated with wallet and account provided by caver-js have no effect on the actual Klaytn network.
+You can easily use your Klaytn account when sign transaction or message by using the [Keyring] / [wallet] provided by caver-js.
```
Comment (kjhman21) [inline README.md:76]:
```suggestion
You can easily use your Klaytn account when signing a transaction or message by using the [Keyring] / [wallet].
```

File: README.md [hunk 2]
```diff
@@ -67,112 +67,79 @@ $ node
...
-[caver.klay.accounts.wallet](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.klay.accounts#wallet) provides the **in-memory wallet** for easy account management in caver-js.
+## Using caver-js keyring/wallet
 
-**Note** Functions associated with wallet and account provided by caver-js have no effect on the actual Klaytn network.
+You can easily use your Klaytn account when sign transaction or message by using the [Keyring] / [wallet] provided by caver-js. 
 
-Let's create a random account as shown in the example below:
-```
-> const account = caver.klay.accounts.create()
+Keyring type ([SingleKeyring], [MultipleKeyring] or [RoleBasedKeyring]) is different according to the type of private key that the user wants to use in caver-js. See [caver.wallet.keyring] for details.
```
Comment (kjhman21) [inline README.md:78]:
```suggestion
The keyring is a new feature that contains an address and one or more private keys based on the key types ([SingleKeyring], [MultipleKeyring] or [RoleBasedKeyring]). Refer to [caver.wallet.keyring] for details.
```

File: README.md [hunk 3]
```diff
@@ -67,112 +67,79 @@ $ node
...
+[caver.wallet] provides the **in-memory wallet** for easy keyring management in caver-js.
```
Comment (kjhman21) [inline README.md:79]:
```suggestion
[caver.wallet], the **in-memory wallet**, is provided to easily manage multiple keyrings.
```

File: README.md [hunk 4]
```diff
@@ -67,112 +67,79 @@ $ node
...
 
-> account
-{ address: '0xF5b66670135666F273F6b5a2eA706A5aCf38D8D5',
-  privateKey: '0x{private key}',
-  ... }
-```
+**Note** Functions associated with wallet and keyring provided by caver-js have no effect on the actual Klaytn blockchain platform (Klaytn).
```
Comment (kjhman21) [inline README.md:81]:
```suggestion
**Note** Functions associated with `wallet` and `keyring` have no effect on the Klaytn blockchain platform. It just manipulates keyrings in the in-memory wallet.
```

File: README.md [hunk 5]
```diff
@@ -67,112 +67,79 @@ $ node
...
 ```
 
-caver-js supports two types of private key formats.
-One is a raw private key format of a 32-byte string type and the other is the [KlaytnWalletKey](https://docs.klaytn.com/klaytn/design/accounts#klaytn-wallet-key-format).
-
-You can also add your account using the KlaytnWalletKey format as shown below:
-```
-// Adding a Klaytn wallet key
-> caver.klay.accounts.wallet.add('0x{private key}0x000x{address in hex}')
+You can add to the wallet instance of caver-js using the keyring object created in the above example, or you can add a keyring using an address and private key(s).
```
Comment (kjhman21) [inline README.md:94]:
```suggestion
You can add the keyring object created in the above example to the wallet instance, or you can add a keyring using an address and private key(s).
```

Comment (kjhman21) [reply README.md]: @jimni1222 음 위 문장대로 하면 아래 예시에서 `keyring`을 넣는 예제가 있어야 할 것 같아요.

> caver.wallet.add(keyring)

요런식인가요?

File: README.md [hunk 6]
```diff
@@ -67,112 +67,79 @@ $ node
...
 ```
+// Adding a keyring to wallet with an address and a private key
```
Comment (kjhman21) [inline README.md:96]:
```suggestion
// Adding a keyring to caver.wallet with an address and a private key
```

File: README.md [hunk 7]
```diff
@@ -67,112 +67,79 @@ $ node
...
+> caver.wallet.newKeyring('0x{address in hex}', '0x{private key}')
 
-Once added to a wallet, it can be accessed via an index or an address.
-```
-> caver.klay.accounts.wallet[0]
-> caver.klay.accounts.wallet['0xF5b66670135666F273F6b5a2eA706A5aCf38D8D5']
-```
+// Adding a keyring to wallet with an address and private keys
```
Comment (kjhman21) [inline README.md:99]:
```suggestion
// Adding a keyring to caver.wallet with an address and private keys
```

File: README.md [hunk 8]
```diff
@@ -67,112 +67,79 @@ $ node
...
+> caver.wallet.newKeyring('0x{address in hex}', ['0x{private key1}', '0x{private key2}', ...])
 
-The private key that matches a specific account stored in the wallet instance can be updated as follows:
-```
-> caver.klay.accounts.wallet.updatePrivateKey('0x{new private key}', '0x{address in hex}')
+// Adding a keyring to wallet with an address and private keys by roles
```
Comment (kjhman21) [inline README.md:102]:
```suggestion
// Adding a keyring to caver.wallet with an address and private keys by roles
```

File: README.md [hunk 9]
```diff
@@ -67,112 +67,79 @@ $ node
...
-> caver.klay.sendTransaction({
-    type: 'VALUE_TRANSFER',
-    from: '0x76d1cc1cdb081de8627cab2c074f02ebc7bce0d0',
-    to: '0x80c2c57dad6cb16488b4c70c17d77152c74f8ade',
-    gas: '300000',
-    value: caver.utils.toPeb('1', 'KLAY'),
-  }).then(console.log)
+You can use caver-js to submit various types of transactions to a node. Please refer to the [caver.transaction](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.transaction/#class) to see how to create a transaction of each type.
+
+You can sign the transaction using a keyring and send a signed transaction through `caver.rpc.klay.sendRawTransaction` as shown below, and the result can be confirmed by the returned receipt:
```
Comment (kjhman21) [inline README.md:109]:
```suggestion
You can sign the transaction using a keyring and send a signed transaction through `caver.rpc.klay.sendRawTransaction` as shown below, and the receipt is returned as a result.
```

File: README.md [hunk 10]
```diff
@@ -67,112 +67,79 @@ $ node
...
 
-// using the event emitter
-> caver.klay.sendTransaction({
-    type: 'VALUE_TRANSFER',
-    from: '0x76d1cc1cdb081de8627cab2c074f02ebc7bce0d0',
-    to: '0x80c2c57dad6cb16488b4c70c17d77152c74f8ade',
-    gas: '300000',
-    value: caver.utils.toPeb('1', 'KLAY'),
-  }).on('transactionHash', function(hash){
+The above example is using `Promise` when send signed transaction to the Klaytn. You can use `event emitter` like below.
```
Comment (kjhman21) [inline README.md:134]:
```suggestion
The above example uses `Promise` when sending a signed transaction to the Klaytn. You can also use `event emitter` like below.
```

File: README.md (hunk 5 above)
Comment (jimni1222) [reply README.md]: 밑에 예제에 주석과 함께 추가했습니다.

File: README.md [hunk 11]
```diff
@@ -67,112 +67,82 @@ $ node
...
-caver-js supports two types of private key formats.
-One is a raw private key format of a 32-byte string type and the other is the [KlaytnWalletKey](https://docs.klaytn.com/klaytn/design/accounts#klaytn-wallet-key-format).
-
-You can also add your account using the KlaytnWalletKey format as shown below:
-```
-// Adding a Klaytn wallet key
-> caver.klay.accounts.wallet.add('0x{private key}0x000x{address in hex}')
+You can add the keyring object created in the above example to the caver.wallet, or you can add a keyring using an address and private key(s).
 ```
+// Adding a keyring instance to caver.wallet
```
Comment (kjhman21) [inline README.md:96]:
```suggestion
// Add a keyring instance to caver.wallet
```

아래 예제와 통일성을 유지하면 좋겠어요. 여기 아래 모두 "Adding" 대신 "Add"로 부탁드릴게요.

Comment (jimni1222) [reply README.md]: 네 반영했습니다

File: README.md [hunk 12]
```diff
@@ -214,32 +185,18 @@ API Specification
...
+* [caver.rpc.klay](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.rpc/klay)
+* [caver.rpc.net](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.rpc/net)
+* [caver.contract](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.contract)
+* [caver.abi](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.abi)
+* [caver.kct.kip7](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.kct/kip7)
+* [caver.kct.kip17](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.kct/kip17)
 * [caver.utils](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.utils)
 
-
-Web3.js Similarity
```
Comment (kjhman21) [inline README.md:225]: 요건 계속 유지해야 하지 않을까요? "web3.js에서 쉽게 caver로 넘어올 수 있어"라는 메시지는 여전히 필요하고, 기능도 가능하니까요..?

Comment (jimni1222) [reply README.md:225]: 밑에 klay rpc 쓰는 부분 수정해서 내용 수정하겠습니다.

**Pull request merged**